contracts/staking/ServiceStakingBase.sol

```diff
@@ -352,16 +352,6 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
         uint256[][] memory serviceNonces
     )
     {
-        // Get the service Ids set length
-        uint256 size = setServiceIds.length;
-        serviceIds = new uint256[](size);
-
-        // Record service Ids
-        for (uint256 i = 0; i < size; ++i) {
-            // Get current service Id
-            serviceIds[i] = setServiceIds[i];
-        }
-
```

Comment:
Don't allocate `serviceIds` if the rewards calculation is not going to be successful.